int64 pkg_id
bool dir